packages/buidler-core/src/builtin-tasks/node.ts

```diff
@@ -63,6 +72,66 @@ Private Key: ${privateKey}
 }
 
 export default function () {
+  internalTask(TASK_NODE_WATCH_COMPILER_OUTPUT)
+    .addParam("url", "The URL of the node", undefined, types.string)
+    .setAction(async ({ url }: { url: string }, { config }) => {
+      const { default: fetch } = await import("node-fetch");
+      const chokidar = await import("chokidar");
+
+      const compilerVersion = config.solc.version;
+      const solcInputPath = path.join(config.paths.cache, SOLC_INPUT_FILENAME);
+      const solcOutputPath = path.join(
+        config.paths.cache,
+        SOLC_OUTPUT_FILENAME
+      );
+
+      const addCompilationResult = async () => {
+        if (
+          !(await fsExtra.pathExists(
+            path.join(config.paths.cache, SOLC_INPUT_FILENAME)
+          ))
+        ) {
+          return false;
+        }
+
+        if (
+          !(await fsExtra.pathExists(
+            path.join(config.paths.cache, SOLC_OUTPUT_FILENAME)
+          ))
+        ) {
+          return false;
+        }
+
+        const compilerInput = await fsExtra.readJSON(solcInputPath, {
```

Comment:
I think these `readJSON` calls can fail. Those fails may have incomplete jsons. 

I know there's `awaitWriteFinish`, but that waits for a single write operation to finish, and I'm not sure if these jsons are written in a single FS operation tbh.

Reply:
No, they are not. I forgot to mention this in the comments.

I think this will work for now, but it has an implicit dependency with how the solc jsons are being written. Right now they are written sequentially, with the input being written first. If that changes, chances are this will throw.

I'm not sure what's the best thing to do here. I think that just adding one or two retries with some delay before failing should cover a lot of ground. What do you think?

Reply:
I'm not sure we need the retries, as the next file modification would trigger this again, right?

> I think this will work for now, but it has an implicit dependency with how the solc jsons are being written. Right now they are written sequentially, with the input being written first. If that changes, chances are this will throw.

Not sure if I followed. I meant multiple `write` syscalls per file. If that happens, this callback may be run, while the sons are incomplete.

Reply:
Oh, I see what you mean. I think `awaitWriteFinish` actually watches the file until its size doesn't change, so that should account for that problem, right?

Reply:
Oh, makes sense.

I think I found another issue though. What happens if an input json is written, and this callback fires before the output json is written? 🤯

Reply:
That's what I was mentioning before: right now `chokidar` is only watching the `output` file. I chose that one because I "know" it's written after the input file. But this is brittle!

I guess a proper solution is to watch both files, and combine and throttle both events. It's somewhat more complex, but maybe it's worth it.

Reply:
Ohh I see. I think we are fine for now. We should have a better strategy when we refactor de compilation pipeline.